### PR TITLE
Week1

### PR DESCRIPTION
**Pull request by AlenoJankowsky**
PR: #2

### CODE REVIEW

File: mario_1_1.c [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        for (int i = 0; i < n; i++)
+
+        {
+            int k = (n - i);
+
+            do
+            {
+                printf("%c", WHITE);
+                k--;
+
```
Comment (matthiasdanne) [inline mario_1_1.c:24]:
```suggestion
```

File: mario_1_1.c [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            {
+                printf("%c", HASH);
+                a++;
+            }
+            while (a <= i);
+            printf("\n");
+        }
+
+
+}
```
Comment (matthiasdanne) [inline mario_1_1.c:41]: As this red sign indicates, a newline is missing at the end of this file.

File: mario_1_2.c [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                j++;
+            }
+            while (j <= i);
+            printf("\n");
+
+
+        }
+
+
+}
```
Comment (matthiasdanne) [inline mario_1_2.c:53]: As this red sign indicates, a newline is missing at the end of this file.

File: mario_1_2.c [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        for (int i = 0; i < n; i++)
+
+        {
+            int k = (n - i);
+
+            do
+            {
+                printf("%c", WHITE);
+                k--;
+
```
Comment (matthiasdanne) [inline mario_1_2.c:24]:
```suggestion
```

File: mario_1_1.c [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+#include <stdio.h>
+
+int main(void)
+{
+const char HASH = '#';
+const char WHITE = ' ';
+int n;
+    do
+    {
+        n = get_int("Height: ");
```
Comment (matthiasdanne) [inline mario_1_1.c:11]: You should add a hint that indicates valid entries.

```suggestion
        n = get_int("Height: (between 1 and 8)");
```

File: mario_1_2.c [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+#include <stdio.h>
+
+int main(void)
+{
+const char HASH = '#';
+const char WHITE = ' ';
+int n;
+    do
+    {
+        n = get_int("Height: ");
```
Comment (matthiasdanne) [inline mario_1_2.c:11]: See above

File: mario_1_2.c [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+            {
+                printf("%c", HASH);
+                j++;
+            }
+            while (j <= i);
+            printf("\n");
+
+
+        }
+
```
Comment (matthiasdanne) [inline mario_1_2.c:51]: I don't know the coding conventionf for C, but one newline at the end of the for-loop and the main function should be enough.

```suggestion
        }
```

**Review by matthiasdanne — CHANGES_REQUESTED** [review]

**Review by matthiasdanne — COMMENTED** [review]
Besides the requested changes, its working as expected. 👍

File: README.md [hunk 8]
```diff
@@ -1,2 +1,3 @@
 # CS50-Harvard
-hallöchen popöchen
\ No newline at end of file
+hallöchen popöchen
```
Comment (DevBastiS) [inline README.md:2]:
```suggestion
```

File: cash.c [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+    int nickels = calculate_nickels(cents);
+    cents = cents - nickels * 5;
+    // Calculate the number of pennies to give the customer
+    int pennies = calculate_pennies(cents);
+    cents = cents - pennies * 1;
+    // Sum coins
+    int coins = quarters + dimes + nickels + pennies;
+    // Print total number of coins to give the customer
+    printf("%i\n", coins);
+}
```
Comment (DevBastiS) [inline cash.c:29]:
```suggestion
}

```

File: cash.c [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+int get_cents(void) {
+    int cents;
+    do
+    {
+        cents = get_int("Change owed: ");
+    }
+    while (cents < 0);
```
Comment (DevBastiS) [inline cash.c:36]:
```suggestion
    while (cents < 0);
    
```

Rule of thumb: Always add a blank line before a return statement. Except it is the only statement in a block.

File: cash.c [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+    return cents;
+}
+int calculate_quarters(int cents) {
+    int n;
+    n = cents / 25;
```
Comment (DevBastiS) [inline cash.c:41]:
```suggestion
    n = cents / 25;
    
```

File: cash.c [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+    return n;
+}
```
Comment (DevBastiS) [inline cash.c:43]:
```suggestion
}

```

File: cash.c [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+int calculate_dimes(int cents) {
+    int n;
+    n = cents / 10;
```
Comment (DevBastiS) [inline cash.c:46]:
```suggestion
    n = cents / 10;
    
```

File: cash.c [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+    return n;
+}
```
Comment (DevBastiS) [inline cash.c:48]:
```suggestion
}

```

File: cash.c [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+int calculate_nickels(int cents) { 
+    int n;
+    n = cents / 5;
+    return n;
+}
```
Comment (DevBastiS) [inline cash.c:53]:
```suggestion
}

```

File: cash.c [hunk 16]
```diff
@@ -0,0 +1,58 @@
...
+int calculate_dimes(int cents) {
+    int n;
+    n = cents / 10;
+    return n;
+}
+int calculate_nickels(int cents) { 
+    int n;
+    n = cents / 5;
```
Comment (DevBastiS) [inline cash.c:51]:
```suggestion
    n = cents / 5;
    
```

File: cash.c [hunk 17]
```diff
@@ -0,0 +1,58 @@
...
+    return n;
+}
+int calculate_pennies(int cents) {
+    int n;
+    n = cents;
```
Comment (DevBastiS) [inline cash.c:56]:
```suggestion
    n = cents;
    
```

File: cash.c [hunk 18]
```diff
@@ -0,0 +1,58 @@
...
+    return n;
```
Comment (DevBastiS) [inline cash.c:57]:
```suggestion
    return cents;
```

File: credit.c [hunk 19]
```diff
@@ -0,0 +1,103 @@
+#include <cs50.h>
+#include <stdio.h>
```
Comment (DevBastiS) [inline credit.c:2]:
```suggestion
#include <stdio.h>

```

File: credit.c [hunk 20]
```diff
@@ -0,0 +1,103 @@
...
+    //The variable sumOFNotMultiplier should add all numbers which weren't used for multiplying
+    long sumOfNotMultiplier = 0;
+    //Check number as descripted in Luhn's Algorithm
+    long check;
+    bool creditCardIsValid = false;
+    while (!creditCardIsValid) {
+        do {
+            n = get_long("please enter your credit card number: \n");
+        }
+        while (n < 1);
```
Comment (DevBastiS) [inline credit.c:19]:
```suggestion
        while (n < 1);
        
```

File: credit.c [hunk 21]
```diff
@@ -0,0 +1,103 @@
...
+        long save_n = n;
+        long save_n_2 = n;
+        long save_n_3 = n;
+        long save_n_4 = n;
+        long save_n_5 = n;
+        n = n / 10;
```
Comment (DevBastiS) [inline credit.c:25]:
```suggestion
        n = n / 10;
        
```

File: credit.c [hunk 22]
```diff
@@ -0,0 +1,103 @@
...
+        do {
+            productSum = productSum + (n % 10) * 2;
```
Comment (DevBastiS) [inline credit.c:27]:
```suggestion
            productSum = productSum + (n % 10) * 2;
            
```

File: credit.c [hunk 23]
```diff
@@ -0,0 +1,103 @@
...
+        while (n > 0 || (n / 10) > 10);
+
+        do {
+            sumOfNotMultiplier = sumOfNotMultiplier + (save_n % 10);
+            save_n = save_n / 100;
+        }
+        while(save_n > 0 || (save_n / 100) >= 10);
+
+        check = sumOfNotMultiplier + productDigitSum;
+        if (check % 10 == 0) {
```
Comment (DevBastiS) [inline credit.c:45]: Ok I have like no idea what is happening in this block :D

Please try the following:
Add speaking variables: 
If adding an if statement think about a good name for the condition. (Example: "IsMasterCard") Then move the condition to the variable and use the variable in the if statement (Example: if (IsMasterCard))

Try to lessen the amount of structures needed.
Decrease the amount of ifs, else ifs, whiles and do-whiles.

File: mario_1_1.c [hunk 24]
```diff
@@ -0,0 +1,41 @@
+#include <cs50.h>
+#include <stdio.h>
+
+int main(void)
+{
```
Comment (DevBastiS) [inline mario_1_1.c:5]: Here the braces are in a new line, in the file above they are in the same line as the statement. Please choose one method and stick to it ;)

File: mario_1_1.c [hunk 25]
```diff
@@ -0,0 +1,41 @@
...
+const char HASH = '#';
+const char WHITE = ' ';
```
Comment (DevBastiS) [inline mario_1_1.c:7]:
```suggestion
const char WHITE = ' ';

```

File: mario_1_1.c [hunk 26]
```diff
@@ -0,0 +1,41 @@
...
+int n;
+    do
+    {
+        n = get_int("Height: ");
+    }
+    while (n < 1 || n > 8);
+
+        for (int i = 0; i < n; i++)
+
```
Comment (DevBastiS) [inline mario_1_1.c:16]:
```suggestion
```

File: mario_1_1.c [hunk 27]
```diff
@@ -0,0 +1,41 @@
...
+            do
+            {
+                printf("%c", HASH);
+                a++;
+            }
+            while (a <= i);
+            printf("\n");
+        }
+
+
```
Comment (DevBastiS) [inline mario_1_1.c:40]:
```suggestion
```

File: mario_1_2.c [hunk 28]
```diff
@@ -0,0 +1,53 @@
+#include <cs50.h>
+#include <stdio.h>
+
+int main(void)
+{
+const char HASH = '#';
+const char WHITE = ' ';
```
Comment (DevBastiS) [inline mario_1_2.c:7]:
```suggestion
const char WHITE = ' ';

```

File: mario_1_2.c [hunk 29]
```diff
@@ -0,0 +1,53 @@
...
+int n;
+    do
+    {
+        n = get_int("Height: ");
+    }
+    while (n < 1 || n > 8);
+
+        for (int i = 0; i < n; i++)
+
```
Comment (DevBastiS) [inline mario_1_2.c:16]:
```suggestion
```

File: mario_1_2.c [hunk 30]
```diff
@@ -0,0 +1,53 @@
...
+        {
+            int k = (n - i);
+
+            do
+            {
+                printf("%c", WHITE);
+                k--;
+
+            }
+
```
Comment (DevBastiS) [inline mario_1_2.c:26]:
```suggestion
```

File: mario_1_2.c [hunk 31]
```diff
@@ -0,0 +1,53 @@
...
+            while (k >=0);
+
+
```
Comment (DevBastiS) [inline mario_1_2.c:29]:
```suggestion
```

File: mario_1_2.c [hunk 32]
```diff
@@ -0,0 +1,53 @@
...
+            int j = 0;
+            do
+            {
+                printf("%c", HASH);
+                j++;
+            }
+            while (j <= i);
+            printf("\n");
+
+
```
Comment (DevBastiS) [inline mario_1_2.c:49]:
```suggestion
```

**Review by DevBastiS — CHANGES_REQUESTED** [review]

File: cash.c [hunk 33]
```diff
@@ -0,0 +1,66 @@
...
+#include <stdio.h>
+
+int get_cents(void);
+int calculate_quarters(int cents);
+int calculate_dimes(int cents);
+int calculate_nickels(int cents);
+int calculate_pennies(int cents);
+
+int main(void) {
+    // Ask how many cents the customer is owed
```
Comment (DevBastiS) [inline cash.c:11]: Remove comments ;)

File: mario_1_2.c (hunk 3 above)
Comment (DevBastiS) [inline mario_1_2.c:53]:
```suggestion
}

```

File: mario_1_1.c (hunk 2 above)
Comment (DevBastiS) [inline mario_1_1.c:41]:
```suggestion
}

```

**Review by DevBastiS — CHANGES_REQUESTED** [review]
Nearly done ;)

File: cash.c [hunk 34]
```diff
@@ -0,0 +1,64 @@
...
+
+int calculate_nickels(int cents) { 
+    int n;
+    n = cents / 5;
+
+    return n;
+}
+
+int calculate_pennies(int cents) {
+
```
Comment (DevBastiS) [inline cash.c:62]:
```suggestion
```

**Review by DevBastiS — APPROVED** [review]

**Review by matthiasdanne — APPROVED** [review]

**Comment by AlenoJankowsky** [comment]
Thank you for your reviews :)

**Pull request merged**